pkgs/development/haskell-modules/configuration-ghc-8.4.x.nix

```diff
@@ -41,4 +41,1041 @@ self: super: {
+  });
+
+  tasty = overrideCabal super.tasty (drv: {
+    ## On Hackage, awaiting for import
+    ## 
+    ##     • Could not deduce (Semigroup (Traversal f))
+    ##         arising from the superclasses of an instance declaration
+    ##       from the context: Applicative f
+    version         = "1.0.0.1";
+    sha256          = "0ggqffw9kbb6nlq1pplk131qzxndqqzqyf4s2p7576nljx11a7qf";
+  });
+
+  test-framework = overrideCabal super.test-framework (drv: {
+    ## On Hackage, awaiting for import
+    ## 
+    ## Setup: Encountered missing dependencies:
+    ## time >=1.1.2 && <1.6
+    version         = "0.8.2.0";
+    sha256          = "1hhacrzam6b8f10hyldmjw8pb7frdxh04rfg3farxcxwbnhwgbpm";
+    ## Setup: Encountered missing dependencies:
+    ## HUnit >=1.2,
+    ## QuickCheck >=2.3 && <2.10,
+    doCheck         = false;
+    ## curl: (22) The requested URL returned error: 404 Not Found
+    ## error: cannot download test-framework-0.8.2.0-r3.cabal from any mirror
+    editedCabalFile = null;
+  });
+
+  xml-conduit = overrideCabal super.xml-conduit (drv: {
+    ## On Hackage, awaiting for import
+    ## 
+    ##     • No instance for (Semigroup Attributes)
+    ##         arising from the superclasses of an instance declaration
+    ##     • In the instance declaration for ‘Monoid Attributes’
+    version         = "1.7.1.0";
+    sha256          = "1c4ip76qgqjdyf77h97mf3yxdimv7m5ma5v20wchn9qjmbkr8ffa";
+  });
+
+  yaml = overrideCabal super.yaml (drv: {
+    ## On Hackage, awaiting for import
+    ## 
+    ##     • No instance for (Semigroup (YamlParser a))
+    ##         arising from the superclasses of an instance declaration
+    ##     • In the instance declaration for ‘Monoid (YamlParser a)’
+    version         = "0.8.28";
+    sha256          = "0swgkzkfrwj0ac7lssn8rnrdfmh3lcsdn5fbq2iwv55di6jbc0pp";
+  });
+
+
+  ## Upstreamed
+
+  constraints = overrideCabal super.constraints (drv: {
+    ## Upstreamed, awaiting a Hackage release
+    ## 
+    ##     • Could not deduce (Semigroup (Dict a))
+    ##         arising from the superclasses of an instance declaration
+    ##       from the context: a
+    src = pkgs.fetchgit {
+      url    = "https://github.com/ekmett/constraints";
+      rev    = "07e8ace4d4a842cf450b346ccc75829bd310f4a5";
+      sha256 = "11l1hvnvnmzd3zxwddz5di4h5iql47gbwjz73x2kdy21avr85i4v";
+    };
+    ## Setup: Encountered missing dependencies:
+    ## hspec >=2, semigroups >=0.11 && <0.19
+    libraryHaskellDepends = drv.libraryHaskellDepends ++ [ self.semigroups self.hspec ];
+  });
+
+  free = overrideCabal super.free (drv: {
+    ## Upstreamed, awaiting a Hackage release
+    ## 
+    ##     • Could not deduce (Semigroup (IterT m a))
+    ##         arising from the superclasses of an instance declaration
+    ##       from the context: (Monad m, Monoid a)
+    src = pkgs.fetchgit {
+      url    = "https://github.com/ekmett/free";
+      rev    = "fcefc71ed302f2eaf60f020046bad392338b3109";
```

Comment:
How will this override be updated when a new release version of `free` comes out that supersedes this one?

Reply:
Good question! -- I've added automation to the audit script to catch the override version lagging behind upstream to the update script.  I'll update the override set tonight.

So the full answer is -- there are scripts that:
1. automatically generate set trimming proposals, and
2. audit the current override set for various invariants, like:
- Nixpkgs-supplied packages lagging behind the Hackage-based overrides 
- Upstreamed-overridden version not lagging behind upstream `master`, in terms of Cabal version number
- User-overridden version having a PR supplied and the PR _not_ being merged
- User-overridden version having the same Cabal version as upstream `master`

When I run the audit script and see any warnings, that means actions need to be taken.

Reply:
OK, if you've automated these things, then that's a pretty good solution. :-) It would be great if you could provide a brief step-by-step write-up of how to run those scripts of yours so that I know how to run them, too.

Reply:
My only concern them being such a mess, I'm somewhat ashamed to bring them to light -- this is truly a case of exploratory chaos -- in bash..

That said, I'll close my eyes and still do this -- just give me a bit more time..

Reply:
I've mostly refactored the thing into a tolerable shape.

I guess a small test suite is what remains.